src/App.tsx

```diff
@@ -1,14 +1,85 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
 export const App: React.FC = () => {
+  const API_TODOS_URL =
+    'https://mate-academy.github.io/react_dynamic-list-of-todos/api/todos.json';
+  const API_USERS_URL =
+    'https://mate-academy.github.io/react_dynamic-list-of-todos/api/users.json';
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [inputValue, setInputValue] = useState<string>('');
+  const [status, setStatus] = useState<string>('all');
+  const [targetTodo, setTargetTodo] = useState<Todo | null>(null);
+  const [name, setName] = useState<string>('');
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    fetch(API_TODOS_URL)
+      .then(response => response.json())
+      .then(_todos => {
+        return _todos.filter((todo: Todo) => {
+          const { title } = todo;
+          const titleToLC = title.toLowerCase();
+          const inputToLC = inputValue.toLowerCase();
+
+          return titleToLC.includes(inputToLC);
+        });
+      })
+      .then(_todos => {
+        return _todos.filter((todo: Todo) => {
+          const { completed } = todo;
+
+          if (status === 'active') {
+            return completed === false;
+          }
+
+          if (status === 'completed') {
+            return completed === true;
+          }
+
+          return todo;
+        });
+      })
+      .then(_todos => setTodos(_todos));
+  }, [status, inputValue]);
```

Comment:
The use of `fetch` for loading todos is correct, but ensure that keys for list items in `TodoList` are stable and do not change between renders. Consider using unique identifiers from the data as keys.